assignments/array-methods.js

```diff
@@ -56,28 +56,37 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+let runnerName = runners.forEach((data) => fullName.push(`${data.first_name} ${data.last_name}`));
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map((data) => data.first_name.toUpperCase());
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter((data) => data.shirt_size === 'L');
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((reducer, data) => reducer += data.donation, 0);
```

Comment:
Style suggestion: if you're passing a callback function and some other value(s) as arguments to a function like reduce, it can make it easier to understand if you put parenthesis around the whole function like so:
`runners.reduce(((reducer, data) => reducer += data.donation), 0);`

For me, the closing parenthesis serves as a better visual clue that _that's_ where the function ends. It's up to you, though :smiley: